pallets/funding/src/lib.rs

```diff
@@ -1280,54 +1325,115 @@ pub mod pallet {
 	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
 		fn on_initialize(now: BlockNumberFor<T>) -> Weight {
 			// Get the projects that need to be updated on this block and update them
+			let mut used_weight = Weight::from_parts(0, 0);
 			for (project_id, update_type) in ProjectsToUpdate::<T>::take(now) {
 				match update_type {
 					// EvaluationRound -> AuctionInitializePeriod | EvaluationFailed
 					UpdateType::EvaluationEnd => {
-						unwrap_result_or_skip!(Self::do_evaluation_end(project_id), project_id, |e| e);
+						used_weight = used_weight.saturating_add(
+							unwrap_result_or_skip!(
```

Comment:
If they had different return types then the macro wouldn't work there. I had the compiler complaining about it.
The reason it's a macro and not a function is that we cannot return an error in on_initialize, so all we do in case of an error is emit an event, and call the keyword "continue" which goes to the next iteration of the loop. (keep in mind we set just 1 project per block right now so this won't do much).